### PR TITLE
TINY-10389: Revert manually dispatching dragend event on drop event in firefox

### PR DESCRIPTION
**Pull request by hamza0867**
Related Ticket: TINY-10389

Description of Changes:
*  Revert manually dispatching dragend event on drop event in firefox

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

**Review by HAFRMO — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/util/QuirksFirefoxTest.ts [hunk 1]
```diff
@@ -80,11 +81,21 @@ describe('browser.tinymce.core.util.QuirksFirefoxTest', () => {
 
   it('TINY-9694: dragend should fire when drop fires with an image', () => {
     const editor = hook.editor();
-    const transfer = new window.DataTransfer();
-    transfer.setData('text/html', '<img src="test">');
-    editor.dispatch('drop', new window.DragEvent('drop', {
-      dataTransfer: transfer
-    }));
-    assertEvents([ 'drop', 'dragend' ]);
+    editor.setContent(`
```
Comment (spocke) [inline modules/tinymce/src/core/test/ts/browser/util/QuirksFirefoxTest.ts:84]: I don't think it needs to be a table I think you could just have two paragraphs and drag/drop between them.

File: modules/tinymce/src/core/main/ts/util/Quirks.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ import * as Rtc from '../Rtc';
  */
 
 interface Quirks {
-  refreshContentEditable (): void;
-  isHidden (): boolean;
+  refreshContentEditable(): void;
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/util/Quirks.ts:23]: Why was these changed? Auto format?

File: .changes/unreleased/tinymce-TINY-10389-2024-01-23.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Removed
```
Comment (spocke) [inline .changes/unreleased/tinymce-TINY-10389-2024-01-23.yaml:2]: I think this should be `Fixed` since `Removed` is more about removing APIs etc more radical changes this removal seems more safe since no one should be dependent on us firing two events here.

**Review by spocke — APPROVED** [review]

**Review by HAFRMO — APPROVED** [review]
